File: ExtremeStudio/DockPanel Forms/MainForm Docks/EditorDock/EditorFunctions/GotoForm.vb
Public Class GotoForm
    Private isClosing as Boolean = False
    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        isClosing = True
        Me.Close()
    End Sub

    Private Sub linenumberRadio_CheckedChanged(sender As Object, e As EventArgs) Handles linenumberRadio.CheckedChanged
        theLabel.Text = "Line Number: "
        maxLabel.Text = "Maximum: " + MainForm.CurrentScintilla?.Lines.Count.ToString()
        curLabel.Text = "Current: " + (MainForm.CurrentScintilla?.CurrentLine + 1).ToString()
        valueTextBox.Select()
    End Sub

    Private Sub positionRadio_CheckedChanged(sender As Object, e As EventArgs) Handles positionRadio.CheckedChanged
        theLabel.Text = "Position: "
        maxLabel.Text = "Maximum: " + MainForm.CurrentScintilla?.Lines(MainForm.CurrentScintilla?.Lines.Count - 1).EndPosition.ToString()
        curLabel.Text = "Current: " + (MainForm.CurrentScintilla?.CurrentPosition + 1).ToString()
        valueTextBox.Select()
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        if valueTextBox.Text = "" Or IsNumeric(valueTextBox.Text) = False Then Exit Sub

        If linenumberRadio.Checked then
            if Convert.ToInt32(valueTextBox.Text) > 0 And Convert.ToInt32(valueTextBox.Text) <= MainForm.CurrentScintilla.Lines.Count then
                MainForm.CurrentScintilla.Lines(Convert.ToInt32(valueTextBox.Text) - 1).Goto()
                isClosing = True
                Close()

            Else
                MsgBox("Please check your value, It seems to be wrong.")
            End If 

        Else If positionRadio.Checked Then
                if Convert.ToInt32(valueTextBox.Text) > 0 And Convert.ToInt32(valueTextBox.Text) <= MainForm.CurrentScintilla.Lines(MainForm.CurrentScintilla.Lines.Count - 1).EndPosition then
                MainForm.CurrentScintilla.GotoPosition(Convert.ToInt32(valueTextBox.Text) - 1)
                isClosing = True
                Close()
                
            Else
                MsgBox("Please check your value, It seems to be wrong.")
            End If

        End If
    End Sub

    Protected Overrides Function ProcessCmdKey(Byref msg As Message, keyData As Keys) As Boolean
        If keyData = Keys.Escape Then
            isClosing = True
            Close()
        End If

        Return MyBase.ProcessCmdKey(msg, keyData)
    End Function

    Private Sub GotoForm_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        linenumberRadio_CheckedChanged(Me, Nothing)
    End Sub

    Private Sub GotoForm_Activated(sender As Object, e As EventArgs) Handles MyBase.Activated
        Opacity = 1.00
    End Sub
    Private Sub GotoForm_Deactivate(sender As Object, e As EventArgs) Handles MyBase.Deactivate
        If isClosing = False Then Opacity = 0.50
    End Sub

    Private Sub GotoForm_FormClosing(sender As Object, e As FormClosingEventArgs) Handles MyBase.FormClosing
        isClosing = True
    End Sub
End Class

File: ExtremeStudio/DockPanelForms/MainFormDocks/EditorDock/EditorFunctions/GotoForm.cs
using System;
using System.Windows.Forms;
using Resources;

namespace ExtremeStudio.DockPanelForms.MainFormDocks.EditorDock.EditorFunctions
{
    public partial class GotoForm : Form
    {
        public GotoForm()
        {
            InitializeComponent();
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            Visible = false;
        }

        private void linenumberRadio_CheckedChanged(object sender, EventArgs e)
        {
            theLabel.Text = translations.GotoForm_linenumberRadio_CheckedChanged_LineNumberLabel;
            maxLabel.Text = translations.GotoForm_linenumberRadio_CheckedChanged_MaximumLabel +
                            Program.MainForm.CurrentScintilla?.Lines.Count;
            curLabel.Text = translations.GotoForm_linenumberRadio_CheckedChanged_CurrentLabel +
                            (Program.MainForm.CurrentScintilla?.CurrentLine + 1);
            valueTextBox.Select();
        }

        private void positionRadio_CheckedChanged(object sender, EventArgs e)
        {
            theLabel.Text = translations.GotoForm_positionRadio_CheckedChanged_PositionLabel;
            maxLabel.Text = translations.GotoForm_linenumberRadio_CheckedChanged_MaximumLabel + Program.MainForm
                                .CurrentScintilla?.Lines[Program.MainForm.CurrentScintilla.Lines.Count - 1].EndPosition;
            curLabel.Text = translations.GotoForm_linenumberRadio_CheckedChanged_CurrentLabel +
                            (Program.MainForm.CurrentScintilla?.CurrentPosition + 1);
            valueTextBox.Select();
        }

        private void GoBtn_Click(object sender, EventArgs e)
        {
            if (valueTextBox.Text == "" || int.TryParse(valueTextBox.Text, out int n) == false)
            {
                return;
            }

            if (linenumberRadio.Checked)
            {
                if (Convert.ToInt32(valueTextBox.Text) > 0 &&
                    Convert.ToInt32(valueTextBox.Text) <= Program.MainForm.CurrentScintilla.Lines.Count)
                {
                    Program.MainForm.CurrentScintilla.Lines[Convert.ToInt32(valueTextBox.Text) - 1].Goto();
                    Visible = false;

                }
                else
                {
                    MessageBox.Show(Convert.ToString(translations.GotoForm_GoBtn_Click_CheckYourValues));
                }

            }
            else if (positionRadio.Checked)
            {
                if (Convert.ToInt32(valueTextBox.Text) > 0 && Convert.ToInt32(valueTextBox.Text) <=
                    Program.MainForm.CurrentScintilla.Lines[Program.MainForm.CurrentScintilla.Lines.Count - 1].EndPosition)
                {
                    Program.MainForm.CurrentScintilla.GotoPosition(Convert.ToInt32(valueTextBox.Text) - 1);
                    Visible = false;

                }
                else
                {
                    MessageBox.Show(Convert.ToString(translations.GotoForm_GoBtn_Click_CheckYourValues));
                }

            }
        }

        protected override bool ProcessCmdKey(ref Message msg, Keys keyData)
        {
            if (keyData == Keys.Escape)
            {
                Visible = false;
            }

            return base.ProcessCmdKey(ref msg, keyData);
        }

        private void GotoForm_Load(object sender, EventArgs e)
        {
            linenumberRadio_CheckedChanged(this, null);
        }

        private void GotoForm_Activated(object sender, EventArgs e)
        {
            Opacity = 1.00;
        }

        private void GotoForm_Deactivate(object sender, EventArgs e)
        {
            Opacity = 0.50;
        }

        private void GotoForm_FormClosing(object sender, FormClosingEventArgs e)
        {
            e.Cancel = true;
            Visible = false;
        }
    }
}